app/components/primer/dropdown/menu.rb

```diff
@@ -0,0 +1,99 @@
+        :dark => "dropdown-menu-dark"
+      }.freeze
+
+      DIRECTION_DEFAULT = :se
+      DIRECTION_OPTIONS = [DIRECTION_DEFAULT, :sw, :w, :e, :ne, :s].freeze
+
+      # @param tag [Symbol] <%= one_of(Primer::Dropdown::Menu::Item::TAG_OPTIONS) %>.
+      # @param divider [Boolean] Whether the item is a divider without any function.
+      # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+      renders_many :items, lambda { |divider: false, **system_arguments|
+        Primer::Dropdown::Menu::Item.new(as: @as, divider: divider, **system_arguments)
+      }
+
+      # @param as [Symbol] When `as` is `:list`, wraps the menu in a `<ul>` with a `<li>` for each item.
+      # @param direction [Symbol] <%= one_of(Primer::Dropdown::Menu::DIRECTION_OPTIONS) %>.
+      # @param header [String] Header to be displayed above the menu.
+      # @param system_arguments [Hash] <%= link_to_system_arguments_docs %>
+      def initialize(as: AS_DEFAULT, direction: DIRECTION_DEFAULT, scheme: SCHEME_DEFAULT, header: nil, **system_arguments)
+        @header = header
+        @direction = direction
+        @as = fetch_or_fallback(AS_OPTIONS, as, AS_DEFAULT)
+
+        @system_arguments = system_arguments
+        @system_arguments[:tag] = "details-menu"
+        @system_arguments[:role] = "menu"
+
+        @system_arguments[:classes] = class_names(
+          @system_arguments[:classes],
+          "dropdown-menu",
+          "dropdown-menu-#{fetch_or_fallback(DIRECTION_OPTIONS, direction, DIRECTION_DEFAULT)}",
+          SCHEME_MAPPINGS[fetch_or_fallback(SCHEME_MAPPINGS.keys, scheme, SCHEME_DEFAULT)]
+        )
+      end
+
+      private
+
+      def list?
+        @as == :list
+      end
+
+      # Items to be rendered in the `Dropdown` menu.
+      class Item < Primer::Component
+        TAG_DEFAULT = :a
+        BUTTON_TAGS = [:button, :summary].freeze
+        TAG_OPTIONS = [TAG_DEFAULT, *BUTTON_TAGS].freeze
+
+        def initialize(as:, tag: TAG_DEFAULT, divider: false, **system_arguments)
+          @divider = divider
+          @as = as
+
+          @system_arguments = system_arguments
+          @system_arguments[:tag] = fetch_or_fallback(TAG_OPTIONS, tag, TAG_DEFAULT)
+          @system_arguments[:tag] = :li if list? && divider?
+          @system_arguments[:role] ||= :menuitem
+          @system_arguments[:role] = :separator if divider
+          @system_arguments[:classes] = class_names(
+            @system_arguments[:classes],
+            "dropdown-item" => !divider,
+            "dropdown-divider" => divider
+          )
+        end
+
+        def call
+          component = if BUTTON_TAGS.include?(@system_arguments[:tag])
+                        Primer::ButtonComponent.new(scheme: :link, **@system_arguments)
+                      else
+                        Primer::BaseComponent.new(**@system_arguments)
+                      end
+
+          # divider has no content
+          render(component) if divider?
```

Comment:
What if there is a `divider` _and_ `content`? Do we raise? Is this documented?

Reply:
if someone uses
```rb
c.item(divider: true) { "some content" }
```

the content is ignored. I don't think we need to raise here

Reply:
Do we need to document that behavior anywhere?

Reply:
ah, that gets tricky since the items are menu's slots, I'm not sure how to handle that

Reply:
Maybe in the param docs for `divider`?

Reply:
I added a `With dividers` section to the docs and added a description for it